### PR TITLE
Cherry-pick Piotr's critical fix for TCP proxy

### PR DESCRIPTION
**Pull request by costinm**
Needed for 0.2.10, without it TCP proxy will be broken for mysql and gRPC.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -37,12 +37,14 @@ bind(
     actual = "//external:ssl",
 )
 
-ENVOY_SHA = "b02d8a9b5033a61a7b13d98999f5f00c4b85b6a4"  # Sep 26, 2017 (use github to download tclap instead of sourceforge)
 
-http_archive(
+ENVOY_SHA = "577410987eddf1d8f47439475c8569dcd68eeb04"  # Cherry-pick Oct 20
+# Branched from "b02d8a9b5033a61a7b13d98999f5f00c4b85b6a4"  # Sep 26, 2017 (use github to download tclap instead of sourceforge)
+
+git_repository(
```
Comment (lizan) [inline WORKSPACE]: Use `http_archive`, this is an intended workaround for Cloud Container Builder.

Comment (costinm) [reply WORKSPACE]: This is using istio/envoy fork - is the archive auto-generated ? And what is the issue with GCB ?

Comment (lizan) [reply WORKSPACE]: Yes same URL format works.

https://github.com/istio/envoy/archive/577410987eddf1d8f47439475c8569dcd68eeb04.zip

The root cause: https://github.com/bazelbuild/bazel/issues/3786

**Comment by costinm** [comment]
Good news is that we have a plan to fix it, by having repo pool the deps,
so bazel bug won't affect us in future.

**Comment by costinm** [comment]
Fixed the workspace to use http_archive, Seb is trying to find something for the jenkins image that is out of date ( or we won't be able to make more releases from the branch..)

**Comment by lizan** [comment]
It's a JGit bug actually, but anyway http_archive gives you better performance on CI.

presubmit fails because it runs bazel 0.6.1 and this branch has to run with bazel 0.5.x.

**Comment by rshriram** [comment]
Why are we cherrypicking envoy commits? We might as well fork envoy for Istio and totally customize it for our needs.

**Comment by sebastienvas** [comment]
This is a the release-0.2 branch.

**Comment by costinm** [comment]
Jenkins image was updated to build master, causing presubmit failures. The cherry-picked PR has passed the tests on master.

**Pull request merged**